### ISSUE TITLE
Can't chain generic object indexes

### ISSUE THREAD

**Issue by partap** [jqlang/jq#2512]
**Describe the bug**
When selecting subfields (e.g. `.foo.bar`), using the generic notation (e.g. `.["foo"].["bar"]`) throws a syntax error.

**To Reproduce**
Given the input (sample.json): 
```json
{
  "title": "Title 1",
  "media:content": {
    "media:credits": "media:content.media:credits 1"
  }
}
```
I want to pull .media-content.media:credits to the top level  and relabel the field to "credits" like this:
```sh
$ cat sample.json | jq '{title, credits: .["media:content"].["media:credits"]}'

``` 

*Expected output*
```json 
{
  "title": "Title 1",
  "credits": "media:content.media:credits 1"
}
```

*Actual output*
```
jq: error: syntax error, unexpected '[', expecting FORMAT or QQSTRING_START (Unix shell quoting issues?) at <top-level>, line 1:
{title, credits: .["media:content"].["media:credits"]}
jq: 1 compile error
```

**Environment (please complete the following information):**
 - OS and Version: macOS Ventura 13.1
 - jq version: 1.6

**Additional context**
If the keys can be used in identifier-like syntax, like this, then it works:
(sample2.json)
```json
{
  "title": "Title 1",
  "content": {
    "credits": "content.credits 1",
  },
}
```  

```sh
$ cat sample2.json | jq '{title, credits: .content.credits}'
```

```json
{
  "title": "Title 1",
  "credits": "content.credits 1"
}
```

But even then , it still doesn't work with the generic quoted syntax chained after the first selector
```sh
$ cat sample2.json | jq '{title, credits: .["content"].["credits"]}'  # Error
$ cat sample2.json | jq '{title, credits: .content.["credits"]}'      # Error
$ cat sample2.json | jq '{title, credits: .["content"].credits}'      # Works
```

---
*Addendum:* 
While writing this, I did finally discover a workaround using the pipe operator:
```sh
$ cat sample2.json | jq '{title, credits: .["content"] | .["credits"]}'.            # Works
$ cat sample.json | jq '{title, credits: .["media:content"] | .["media:credits"]}'. # Works
```
But I'm filing the bug report anyways because, according to the docs,  `.foo.bar` *should* be equivalent to `.["foo"].["bar"]`.
(and likewise, `.["foo"].["bar"]` should be equivalent to `.["foo'] | .["bar"]`

Thanks for the tool!

**Comment by pkoppstein** [comment]
You misread the documents.  The good news is that `.foo.bar` can also be written as `.["foo"]["bar"]`, which is one less pesky ".".

**Comment by partap** [comment]
Ah, thanks!

_Although..._

If I misread the documents, I'm _still_ misreading them...Check again:

> For example `.["foo::bar"]` and `.["foo.bar"]` work while `.foo::bar` does not, and `.foo.bar` means `.["foo"].["bar"]`.

https://stedolan.github.io/jq/manual/#Basicfilters

**Comment by pkoppstein** [comment]
@partap - I have edited my remarks for clarity.

**Comment by partap** [comment]
Actually, both versions of documentation show the same syntax...
https://stedolan.github.io/jq/manual/v1.6/#Basicfilters

I was using the v1.6 docs initially but posted the dev version here because I was checking to see if that section had been updated. Looks like the program behavior changed to match the documentation in the latest dev releases?

Anyways, I'm not trying to hassle you about this, just letting you know that the 1.6 docs are misleading. (v1.6 is the current version in Homebrew)

I eventually did find a way to make it work, but it took quite a while and lots of trial and error. Just trying to be helpful for the next person that comes along...
